cmd/fmt.go

```diff
@@ -138,6 +138,10 @@ func formatFile(params *fmtCommandParams, out io.Writer, filename string, info o
 		DropV0Imports: params.dropV0Imports,
 	}
 
+	if params.regoV1 {
+		opts.ParserOptions = &ast.ParserOptions{RegoVersion: ast.RegoV0}
```

Comment:
Since we expect incoming policies to be v0, we'll error if the relevant imports aren't present; which isn't ideal.

If we drop this, then we'll default to v1, and the user can override to with `--v0-compatible` (now they can actually override to v1 with the hidden `--v1-compatible` flag).

Alternatively, we could try parsing with one version, and if it fails, parse with the other; but that's hard to make fool-proof.

Reply:
To clarify the current behaviour:

* `opa fmt`:
  * v1 rego is formatted to v1
  * `rego.v1`/`future.keywords` imports are NOT removed
  * `rego.v1`/`future.keywords` imports are NOT added if missing
  * v0 rego is rejected
* `opa fmt --v0-compatible`:
  * v0 rego is formatted to v0
  * v1 rego is rejected 
* `opa fmt --v0-v1`:
  * v0 rego is formatted to be compatible with v0 AND v1
  * v0 rego is rejected
* `opa fmt --v0-v1 --v1-compatible`:
  * v1 rego is formatted to be compatible with v0 AND v1
  * v1 rego is rejected
  * `--v1-compatible` flag is hidden in `opa fmt --help`

A mix of v0/v1 modules is NOT supported.